src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionMicroserviceVerticle.java

```diff
@@ -965,40 +975,137 @@ private void downloadFileAnnotation(RoutingContext event) {
-                            response.setStatusCode(404);
-                            response.end("File for given Annotation does not exist");
-                            return;
-                        }
-                        if (file.isDirectory()) {
-                            response.setStatusCode(501);
-                            response.end("File Annotation of Unsupported File Type");
-                            return;
-                        }
-                        else {
-                            response.headers().set("Content-Type", "application/octet-stream");
-                            response.headers().set("Content-Disposition",
-                                    "attachment; filename=\"" + fileName + "\"");
-                            response.sendFile(filePath);
+            ImageRegionVerticle.GET_FILE_ANNOTATION_METADATA_EVENT,
+            Json.encode(annotationCtx), new Handler<AsyncResult<Message<JsonObject>>>() {
+                @Override
+                public void handle(AsyncResult<Message<JsonObject>> result) {
+                    if (result.failed()) {
+                        log.error(result.cause().getMessage());
+                        response.setStatusCode(404);
+                        response.end("Could not get annotation "
+                                    + request.getParam("annotationId"));
+                        return;
+                    }
+                    JsonObject fileInfo = result.result().body();
+                    String fileName = fileInfo.getString("originalFileName");
+                    String filePath = fileInfo.getString("originalFilePath");
+                    //If the path is a directory, send error response
+                    File file = new File(filePath);
+                    if (!file.exists()) {
+                        response.setStatusCode(404);
+                        response.end("File for given Annotation does not exist");
+                        return;
+                    }
+                    if (file.isDirectory()) {
+                        response.setStatusCode(501);
+                        response.end("File Annotation of Unsupported File Type");
+                        return;
+                    }
+                    else {
+                        response.headers().set("Content-Type", "application/octet-stream");
+                        response.headers().set("Content-Disposition",
+                                "attachment; filename=\"" + fileName + "\"");
+                        if (request.headers().contains("Range")) {
+                            String range = request.getHeader("Range");
+                            if (range.matches("^bytes=\\d+-\\d+$")) {
+	                            String[] startEndStr =
+	                            		range.substring("bytes=".length()).split("-");
+	                            response.sendFile(filePath, Long.valueOf(startEndStr[0]),
```

Comment:
Not sure if I'm looking at the correct documentation, but `sendFile` takes offset and length apparently, not start and end: https://vertx.io/docs/3.9.16/apidocs/io/vertx/reactivex/core/http/HttpServerResponse.html#sendFile-java.lang.String-